test/unit/protocols/test_gst.py

```diff
@@ -215,18 +217,47 @@ def setUpClass(cls):
         cls.gst_data = ProtocolData(cls.gst_design, ds)
 
 
-class GateSetTomographyTester(BaseProtocolData, BaseCase):
+class MapForwardSimulatorWrapper(MapForwardSimulator):
+
+    Message = """
+        Hit the forward simulator wrapper!
+    """
+
+    def _bulk_fill_probs(self, array_to_fill, layout):
+        print(self.Message)
+        super(MapForwardSimulatorWrapper, self)._bulk_fill_probs(array_to_fill, layout)
+
+    def _bulk_fill_probs_atom(self, array_to_fill, layout_atom, resource_alloc):
+        print(self.Message)
+        super(MapForwardSimulatorWrapper, self)._bulk_fill_probs_atom(array_to_fill, layout_atom, resource_alloc)
+
+
+class TestGateSetTomography(BaseProtocolData):
     """
     Tests for methods in the GateSetTomography class.
+
+    We can't subclass BaseCase since we use some advanced PyTest features.
     """
 
     def test_run(self):
+        self.setUpClass()
         proto = gst.GateSetTomography(smq1Q_XYI.target_model("CPTPLND"), 'stdgaugeopt', name="testGST")
         results = proto.run(self.gst_data)
 
         mdl_result = results.estimates["testGST"].models['stdgaugeopt']
         twoDLogL = two_delta_logl(mdl_result, self.gst_data.dataset)
-        self.assertLessEqual(twoDLogL, 1.0)  # should be near 0 for perfect data
+        assert twoDLogL <= 1.0  # should be near 0 for perfect data
+
+    def test_run_custom_sim(self, capfd: pytest.LogCaptureFixture):
+        self.setUpClass()
+        proto = gst.GateSetTomography(smq1Q_XYI.target_model("CPTPLND"), 'stdgaugeopt', name="testGST")
+        results = proto.run(self.gst_data, simulator=MapForwardSimulatorWrapper())
+        stdout, _ = capfd.readouterr()
+        assert MapForwardSimulatorWrapper.Message in stdout
```

Comment:
Clever. We could also check the type of the simulator in the results models also.

Reply:
Change made in [e66ad37](https://github.com/sandialabs/pyGSTi/pull/358/commits/e66ad376c5dd9106a4be21386650b47e9f13d925).